### PR TITLE
add additional image drawing methods and operators

### PR DESCRIPTION
**Pull request by tteichmeister**
Hi, 

It would be helpful to provide additional drawing functions for the image. 
Especially to fill a rectangle or a circle. Currently it is only possible to draw a rectangle/circle but not to fill it. 

Furthermore, an additional function to draw a text with a background color would be good. 
Especially if you want to label objects and other stuff, with a background color as a contrast, it can help to improve the readability of the text.

(Here I marked the object with `DrawRectangle` and using the new `DrawText` method to draw the text below)
image

Therefore I provided 3 methods: 
- `FillRectangle` 
- `FillCircle`
- `DrawText` (including an additional color for the background color)

I also including the operators which are also included for the existing `DrawText`, `DrawRectangle`...

### CODE REVIEW

File: Sources/Imaging/Microsoft.Psi.Imaging/ImageExtensions.cs [hunk 1]
```diff
@@ -1375,6 +1375,142 @@ public static void DrawText(this Image image, string str, Point p0, Color color
             drawFormat.FormatFlags = 0;
             graphics.DrawString(str, drawFont, drawBrush, p0.X, p0.Y, drawFormat);
         }
+
+        /// <summary>
+        /// Renders text on the image at the specified pixel (p0).
+        /// </summary>
+        /// <param name="image">Image to draw on.</param>
+        /// <param name="str">Text to render.</param>
+        /// <param name="p0">Pixel coordinates for center of circle.</param>
```
Comment (chitsaw) [inline Sources/Imaging/Microsoft.Psi.Imaging/ImageExtensions.cs:1384]: upper-left corner of the text

File: Sources/Imaging/Test.Psi.Imaging.Windows/ImageTester.cs [hunk 2]
```diff
@@ -93,6 +99,38 @@ public void Image_GrayDrawText()
...
+        }
+
+        [TestMethod]
+        [Timeout(60000)]
+        public void Image_GrayFillRect()
+        {
+            using var sharedImage = ImagePool.GetOrCreate(this.testImage_Gray.Width, this.testImage_Gray.Height, this.testImage_Gray.PixelFormat);
+            this.testImage_Gray.CopyTo(sharedImage.Resource);
+            sharedImage.Resource.FillRectangle(new System.Drawing.Rectangle(0, 0, 20, 20), System.Drawing.Color.White);
+            sharedImage.Resource.ToBitmap().Save(@"C:\Users\tteichmeister\Desktop\fill_rect.bmp");
```
Comment (chitsaw) [inline Sources/Imaging/Test.Psi.Imaging.Windows/ImageTester.cs:119]: I think this was temporary and can be removed

File: Sources/Imaging/Test.Psi.Imaging.Windows/ImageTester.cs [hunk 3]
```diff
@@ -93,6 +99,38 @@ public void Image_GrayDrawText()
...
+        }
+
+        [TestMethod]
+        [Timeout(60000)]
+        public void Image_GrayFillCircle()
+        {
+            using var sharedImage = ImagePool.GetOrCreate(this.testImage_Gray.Width, this.testImage_Gray.Height, this.testImage_Gray.PixelFormat);
+            this.testImage_Gray.CopyTo(sharedImage.Resource);
+            sharedImage.Resource.FillCircle(new System.Drawing.Point(this.testImage_Gray.Width / 2, this.testImage_Gray.Height / 2), 100, System.Drawing.Color.White);
+            sharedImage.Resource.ToBitmap().Save(@"C:\Users\tteichmeister\Desktop\fill_circle.bmp");
```
Comment (chitsaw) [inline Sources/Imaging/Test.Psi.Imaging.Windows/ImageTester.cs:130]: remove

**Review by chitsaw — CHANGES_REQUESTED** [review]
Thank you for submitting this PR - just a couple of tiny changes but otherwise everything looks great! Thanks also for including unit tests :-). For some reason there is one test (Image_GrayDrawTestWithBackground) that fails the CI build due to too many different pixels vs the reference image. I wonder if that test passes for you at all? We've sometimes noticed in the past that some of these imaging tests can fail on some machines but not others, possibly due to differences in GDI+ versions. There are `tolerance` and `outlier` parameters in the `AssertAreImagesEqual` method for that reason, but normally the default tolerances have worked for us.

**Comment by tteichmeister** [comment]
> Thank you for submitting this PR - just a couple of tiny changes but otherwise everything looks great! Thanks also for including unit tests :-). For some reason there is one test (Image_GrayDrawTestWithBackground) that fails the CI build due to too many different pixels vs the reference image. I wonder if that test passes for you at all? We've sometimes noticed in the past that some of these imaging tests can fail on some machines but not others, possibly due to differences in GDI+ versions. There are `tolerance` and `outlier` parameters in the `AssertAreImagesEqual` method for that reason, but normally the default tolerances have worked for us.

@chitsaw 
It's strange, because before I submitted it, the Test was working on my machine. When I was doing it now again, this test didn't passed also on my machine. So I provided a new verification image for this test. I think I uploaded the wrong one. Since the checks passed now :-)

Thank you for your review and to let me contribute this to your project :-).

**Review by chitsaw — APPROVED** [review]

**Comment by chitsaw** [comment]
@tteichmeister thanks for making those changes. Yes, the tests pass now with the new image. Thanks so much for your contributions!

**Pull request merged**